### PR TITLE
feat: Updating base Terraform module

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

Updates the base Terraform module for AUSF to the latest agreed form, in particular:
- removes integrations from `main.tf`
- models Juju integrations using `output.tf`
- updates README to show how to use base module (README with information about how to deploy will come along with a higher level module)

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: terraform/modules/terraform-juju-sdcore-ausf-k8s/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+## Module structure
+
+- **main.tf** - Defines the Juju application to be deployed.
+- **variables.tf** - Allows customization of the deployment. Except for exposing the deployment
+  options (Juju model name, channel or application name) also models the charm configuration.
+- **output.tf** - Responsible for integrating the module with other Terraform modules, primarily
+  by defining potential integration endpoints (charm integrations), but also by exposing
+  the application name.
+- **terraform.tf** - Defines the Terraform provider.
+-
```
Comment (gatici) [inline terraform/modules/terraform-juju-sdcore-ausf-k8s/README.md:20]: - needs to removed.

Comment (Gmerold) [reply terraform/modules/terraform-juju-sdcore-ausf-k8s/README.md]: True. Thanks. 
Done.

**Comment by gatici** [comment]
I think we need to get the team opinions' regarding with TF module directory movement to `terraform/modules/terraform-juju-sdcore-ausf-k8s`. Other than that LGTM.

**Comment by Gmerold** [comment]
> I think we need to get the team opinions' regarding with TF module directory movement to `terraform/modules/terraform-juju-sdcore-ausf-k8s`. Other than that LGTM.

Yeah, I figured we'll need to move it anyway when we start working on the module with all the dependencies. We'll need to store it somewhere. I thought we could have it directly under `terraform` folder.

**Review by gatici — APPROVED** [review]

File: terraform/README.md [hunk 2]
```diff
@@ -1,75 +1,56 @@
...
-terraform plan -var-file="terraform.tfvars" 
-```
-
-Deploy the resources:
-
-```console
-terraform apply -auto-approve 
+```text
+module "ausf" {
+  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator/terraform/modules/terraform-juju-sdcore-ausf-k8s"
```
Comment (gatici) [inline terraform/README.md:28]:
```suggestion
  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator//terraform/"
```

File: terraform/README.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ like shown below:
 
 ```text
 module "ausf" {
-  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator/terraform/modules/terraform-juju-sdcore-ausf-k8s"
+  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator/terraform"
```
Comment (gatici) [inline terraform/README.md:28]: https://developer.hashicorp.com/terraform/language/modules/sources#modules-in-package-sub-directories

As the module is located in a repo subdirectory `//terraform` needs to be used according to guide. This is also tested and works in this way.

File: terraform/README.md [hunk 4]
```diff
@@ -1,75 +1,56 @@
...
-terraform plan -var-file="terraform.tfvars" 
-```
-
-Deploy the resources:
-
-```console
-terraform apply -auto-approve 
+```text
+module "ausf" {
+  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator/terraform"
```
Comment (gatici) [inline terraform/README.md:28]:
```suggestion
  source = "git::https://github.com/canonical/sdcore-ausf-k8s-operator//terraform"
```

**Review by gatici — CHANGES_REQUESTED** [review]

File: terraform/README.md (hunk 3 above)
Comment (Gmerold) [reply terraform/README.md:28]: This is only applicable when the module is in a subdirectory of a package. It's not the case here.

**Review by gatici — APPROVED** [review]

**Pull request merged**